contrib/drivers/filesharedrivers/chubaofs/util.go

```diff
@@ -0,0 +1,340 @@
+		errmsg := fmt.Sprintf("chubaofs: %v failed to read http response body, bodyLen(%d) err(%v)", req, len(body), err)
+		return errors.New(errmsg)
+	}
+
+	resp := &generalVolumeResponse{}
+	if err := json.Unmarshal(body, resp); err != nil {
+		errmsg := fmt.Sprintf("chubaofs: %v failed to unmarshal, url(%v) msg(%v)", req, url, resp.Msg)
+		return errors.New(errmsg)
+	}
+
+	if resp.Code != 0 {
+		errmsg := fmt.Sprintf("chubaofs: %v failed, url(%v) code(%v) msg(%v)", req, url, resp.Code, resp.Msg)
+		return errors.New(errmsg)
+	}
+
+	log.Infof("chubaofs: %v url(%v) successful!", req, url)
+	return nil
+}
+
+func doMount(cmdName string, confFile []string) error {
+	env := []string{
+		fmt.Sprintf("PATH=%s", os.Getenv("PATH")),
+	}
+
+	for _, conf := range confFile {
+		cmd := exec.Command(cmdName, "-c", conf)
+		cmd.Env = append(cmd.Env, env...)
+		if msg, err := cmd.CombinedOutput(); err != nil {
+			return errors.New(fmt.Sprintf("chubaofs: failed to start client daemon, msg: %v , err: %v", string(msg), err))
+		}
+	}
+	return nil
+}
+
+func doUmount(mntPoints []string) error {
+	env := []string{
+		fmt.Sprintf("PATH=%s", os.Getenv("PATH")),
+	}
+
+	for _, mnt := range mntPoints {
+		cmd := exec.Command("umount", mnt)
+		cmd.Env = append(cmd.Env, env...)
+		msg, err := cmd.CombinedOutput()
+		if err != nil {
+			return errors.New(fmt.Sprintf("chubaofs: failed to umount, msg: %v , err: %v", msg, err))
+		}
+	}
+	return nil
+}
+
+/*
+ * This function creates client config files according to export locations.
+ */
+func prepareConfigFiles(d *Driver, opt *pb.CreateFileShareOpts) (configFiles, fsMntPoints []string, owner string, err error) {
+	volName := opt.GetId()
+	configFiles = make([]string, 0)
+	fsMntPoints = make([]string, 0)
+
+	/*
+	 * Check client runtime path.
+	 */
+	fi, err := os.Stat(d.conf.ClientPath)
+	if err != nil || !fi.Mode().IsDir() {
+		err = errors.New(fmt.Sprintf("chubaofs: invalid client runtime path, path: %v, err: %v", d.conf.ClientPath, err))
+		return
+	}
+
+	clientConf := path.Join(d.conf.ClientPath, volName, "conf")
+	clientLog := path.Join(d.conf.ClientPath, volName, "log")
+	clientWarnLog := path.Join(d.conf.ClientPath, volName, "warnlog")
+
+	if err = os.MkdirAll(clientConf, os.ModeDir); err != nil {
+		err = errors.New(fmt.Sprintf("chubaofs: failed to create client config dir, path: %v , err: %v", clientConf, err))
+		return
+	}
+	defer func() {
+		if err != nil {
+			os.RemoveAll(clientConf)
+		}
+	}()
+
+	if err = os.MkdirAll(clientLog, os.ModeDir); err != nil {
+		err = errors.New(fmt.Sprintf("chubaofs: failed to create client log dir, path: %v", clientLog))
+		return
+	}
+	defer func() {
+		if err != nil {
+			os.RemoveAll(clientLog)
+		}
+	}()
+
+	if err = os.MkdirAll(clientWarnLog, os.ModeDir); err != nil {
+		err = errors.New(fmt.Sprintf("chubaofs: failed to create client warn log dir, path: %v , err: %v", clientWarnLog, err))
+		return
+	}
+	defer func() {
+		if err != nil {
+			os.RemoveAll(clientWarnLog)
```

Comment:
Can we have some logging before removing warnlog, log or config?

Reply:
Sure, I'll add some warnings here. However, just for clarification, these directories are created during the volume creation process, so upon reaching here, the creation just failed.